src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -0,0 +1,258 @@
+                    currentClass.AddMethod(clone);
+                }
+
+                // Build root command
+                var clientConstructor = currentClass.GetChildElements().OfType<CodeMethod>().FirstOrDefault(m => m.MethodKind == CodeMethodKind.ClientConstructor);
+                if (clientConstructor != null)
+                {
+                    var rootMethod = new CodeMethod
+                    {
+                        Name = "BuildCommand",
+                        IsAsync = false,
+                        MethodKind = CodeMethodKind.CommandBuilder,
+                        ReturnType = new CodeType { Name = "Command", IsExternal = true },
+                        OriginalMethod = clientConstructor,
+                    };
+                    currentClass.AddMethod(rootMethod);
+                }
+            }
+            CrawlTree(currentElement, CreateCommandBuilders);
+        }
+
+        private static CodeType CreateCommandType(CodeElement parent)
+        {
+            return new CodeType
+            {
+                Name = "Command",
+                IsExternal = true
+            };
+        }
+
+        private static CodeMethod CreateBuildCommandMethod(CodeProperty navProperty, CodeClass parent)
+        {
+            var codeMethod = new CodeMethod();
+            codeMethod.IsAsync = false;
+            codeMethod.Name = $"Build{navProperty.Name.ToFirstCharacterUpperCase()}Command";
+            codeMethod.MethodKind = CodeMethodKind.CommandBuilder;
+            codeMethod.ReturnType = CreateCommandType(codeMethod);
+            codeMethod.AccessedProperty = navProperty;
+            codeMethod.SimpleName = navProperty.Name;
+            return codeMethod;
+        }
+
+        private static void AddParsableInheritanceForModelClasses(CodeElement currentElement)
+        {
+            if (currentElement is CodeClass currentClass &&
+                currentClass.IsOfKind(CodeClassKind.Model) &&
+                currentClass.StartBlock is CodeClass.Declaration declaration)
+            {
+                declaration.AddImplements(new CodeType
+                {
+                    IsExternal = true,
+                    Name = $"IParsable",
+                });
+                (currentClass.Parent is CodeClass parentClass &&
+                parentClass.StartBlock is CodeClass.Declaration parentDeclaration ?
+                    parentDeclaration :
+                    declaration)
+                    .AddUsings(new CodeUsing
+                    {
+                        Name = "Microsoft.Kiota.Abstractions.Serialization"
+                    });
+            }
+            CrawlTree(currentElement, AddParsableInheritanceForModelClasses);
+        }
+
+        private static readonly AdditionalUsingEvaluator[] defaultUsingEvaluators = new AdditionalUsingEvaluator[] {
+            new (x => x is CodeProperty prop && prop.IsOfKind(CodePropertyKind.RequestAdapter),
```

Comment:
could you check whether there's a way to call in the array we already have in the CSharp refiner, and add the additional things you need on top of that ? Trying to avoid duplication